tools/load_balancing_tool/load_balancing_submit.py

```diff
@@ -0,0 +1,326 @@
+    for pesize in pesize_list:
+        casename = casename_prefix + pesize
+        casedir = os.path.join(script_dir, casename)
+        logger.info('Case %s...', casename)
+        if os.path.exists(casedir):
+            if force_purge:
+                logger.info('Removing directory %s', casedir)
+                shutil.rmtree(casedir)
+            elif os.path.exists(os.path.join(casedir, 'timing')):
+                logger.info('Skipping timing run for %s, because case '
+                            'already exists.\n   To force rerun of all tests, '
+                            'use --force_purge option.\n   To force rerun of '
+                            'this test only, remove directory \n      %s',
+                            casename, casedir)
+                continue
+            else:
+                logger.critical("ERROR: directory %s exists,\n  but no timing "
+                                "information available. Either run with "
+                                "--force_purge\n  (remove all timing runs) or "
+                                "remove this directory and try again", casedir)
+                sys.exit(1)
+
+        cmd = '%s/create_newcase --case %s --output-root %s ' % \
+              (script_dir, casename, casedir)
+        cmd += create_newcase_flags
+        logger.info(cmd)
+        run_cmd_no_fail(cmd, from_dir=script_dir)
+        for var in CIME_DEFAULTS:
+            _set_xml_val(var, CIME_DEFAULTS[var], casedir)
+
+        if extra_options_file is not None:
+            try:
+                extras = open(extra_options_file, 'r')
+                for line in extras.readlines():
+                    split = line.split('=')
+                    if len(split) == 2:
+                        logger.info('setting %s=%s', split[0], split[1])
+                        _set_xml_val(split[0], split[1], casedir)
+                    else:
+                        logger.debug('ignoring line in %s: %s',
+                                     extra_options_file, line)
+                extras.close()
+            except IOError, e:
+                logger.critical("ERROR: Could not read file %s",
+                                extra_options_file)
+                raise SystemExit(1)
+
+        pes_ntasks, pes_nthrds, pes_rootpe, ignore = \
+            pesobj.find_pes_layout('any', 'any', 'any', pesize_opts=pesize)
+
+        for key in pes_ntasks:
+            _set_xml_val(key, pes_ntasks[key], casedir)
+        for key in pes_nthrds:
+            _set_xml_val(key, pes_nthrds[key], casedir)
+        for key in pes_rootpe:
+            _set_xml_val(key, pes_rootpe[key], casedir)
+
+        cmd = './case.setup'
+        logger.info(cmd)
+        run_cmd_no_fail(cmd, from_dir=casedir)
+
+        cmd = './case.build'
+        logger.info(cmd)
+        run_cmd_no_fail(cmd, from_dir=casedir)
+
+        cmd = './case.submit'
+        logger.info(cmd)
+        run_cmd_no_fail(cmd, from_dir=casedir)
```

Comment:
I think that you should consider using the PFS system test instead of create_newcase - the test will do the setup build and submit steps internally.